addon/components/cf-field/input/powerselect.js

```diff
@@ -0,0 +1,89 @@
+import Component from "@ember/component";
+import { computed } from "@ember/object";
+import { inject as service } from "@ember/service";
+import layout from "../../../templates/components/cf-field/input/powerselect";
+
+/**
+ * Dropdown component for the single and multiple choice question type
+ *
+ * @class CfFieldInputPowerSelectComponent
+ * @argument {Field} field The field for this input type
+ */
+export default Component.extend({
+  layout,
+  tagName: "",
+  intl: service(),
+
+  multiple: computed("field.question.__typename", function() {
+    return this.get("field.question.__typename").startsWith("Multiple");
+  }),
+
+  choices: computed(
+    "multiple",
+    "field.question.{choiceOptions,multipleChoiceOptions}.edges",
+    function() {
+      let options = this.get("multiple")
+        ? this.get("field.question.multipleChoiceOptions")
+        : this.get("field.question.choiceOptions");
+
+      return options.edges.map(edge => edge.node);
+    }
+  ),
+
+  selected: computed(
+    "field.answer.{_valueKey,listValue,stringValue}",
+    function() {
+      let key = this.get("field.answer._valueKey");
+      let path = `field.answer.${key}`;
+      let answers = this.get(path);
+      let selection = null;
+
+      if (answers !== null) {
+        selection = this.get("choices").filter(choice => {
+          return answers.includes(choice.slug);
+        });
+
+        if (key === "stringValue") {
+          selection = selection[0];
+        }
+      }
+
+      return selection;
```

Comment:
I would have written this like that:
```js
      const key = this.get("field.answer._valueKey");
      const answers = this.get(`field.answer.${key}`);

      if (!answers) {
        return null;
      }
      const selection = this.get("choices").filter(choice => {
        return answers.includes(choice.slug);
      });

      return key === "stringValue" ? selection[0] : selection:
```

* prefer-const-over-let (we should add this to our eslint config)
* nest code as little as possible / return early
* as little assignment as possible

Reply:
I personally prefer to have only one return per function where possible. But I can live with multiple returns :-)

What options would you recommend for prefer-const?
https://eslint.org/docs/rules/prefer-const

Reply:
Right - in my opinion the mental overhead of tracking the value of a variable is higher than of finding all the `return` statements in a function. Also we try to write our JS in a "functional" style which means avoiding mutable variables.

See also: https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement

For prefer-const I'd just stick with the default options, resp. not set any specific options.